### PR TITLE
Test:  fix flaky package signing test

### LINKED ISSUE #8218: Test:  NuGet.Packaging.FuncTest.SignedPackageIntegrityVerificationTests.VerifyPackageContentHash_SignedPackagesAsync is flaky
[`NuGet.Packaging.FuncTest.SignedPackageIntegrityVerificationTests.VerifyPackageContentHash_SignedPackagesAsync`](https://github.com/NuGet/NuGet.Client/blob/bdcec93681afdd1f5cd553af07414a3b3d9a866c/test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs#L950) is flaky.

The test [creates package \#1](https://github.com/NuGet/NuGet.Client/blob/bdcec93681afdd1f5cd553af07414a3b3d9a866c/test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs#L957) and generates an expected content hash from it.  Then the test calls [`SignedArchiveTestUtility.AuthorSignPackageAsync(…)`](), which [creates package \#2](https://github.com/NuGet/NuGet.Client/blob/bdcec93681afdd1f5cd553af07414a3b3d9a866c/test/TestUtilities/Test.Utility/Signing/SignedArchiveTestUtility.cs#L49), signs it, and returns a reference to the signed package.  The test then calls `PackageArchiveReader.GetContentHash(…)` on the signed package and compares the expected content hash with the actual content hash.

The issue is that two separate ZIP files created within a very short timeframe are likely to have the same datetime values in local file and central directory headers, but that is not guaranteed.  If the two ZIP files have any differences other than those related to the package signature file, the content hashes will be different.

The fix is to copy the first package for use in author signing the package.

### PR DESCRIPTION
**Pull request by dtivel**
Fix https://github.com/NuGet/Home/issues/8218.

### CODE REVIEW

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs [hunk 1]
```diff
@@ -952,33 +952,30 @@ public async Task VerifyPackageContentHash_SignedPackagesAsync()
...
             var nupkg = new SimpleTestPackageContext();
 
-            string expectedHash = null;
-
-            using (var zipStream = await nupkg.CreateAsStreamAsync())
+            using (MemoryStream packageStream = await nupkg.CreateAsStreamAsync())
             {
-                expectedHash = Convert.ToBase64String(new CryptoHashProvider("SHA512").CalculateHash(zipStream));
-            }
+                string expectedHash = Convert.ToBase64String(new CryptoHashProvider("SHA512").CalculateHash(packageStream));
```
Comment (zivkan) [inline test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs:957]: I've seen cryptographic exceptions in past projects when some crypto related classes were garbage collected without being disposed, I just can't remember which classes had this behaviour. Perhaps we could prevent other flakiness if we make sure the `new CryptoHashProvider` gets disposed as well.

**Review by zivkan — APPROVED** [review]
It's not clear to me how this fixes test flakiness (it copies a nupkg before author signing it?). Otherwise the changes look good to me.

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs (hunk 1 above)
Comment (dtivel) [reply test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs:957]: [`CryptoHashProvider`](https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/src/NuGet.Core/NuGet.Common/CryptoHashProvider.cs#L14) does not implement `IDisposable`, but internally `CryptoHashProvider.CalculateHash(Stream)` already [disposes](https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/src/NuGet.Core/NuGet.Common/CryptoHashProvider.cs#L61) the `HashAlgorithm` instance it creates.

**Comment by dtivel** [comment]
@zivkan, the bug was that the test created one ZIP [here](https://github.com/NuGet/NuGet.Client/pull/2884/files#diff-165cb83f000cbfea0fde91da5236f31cL957) and then a second one [here](https://github.com/NuGet/NuGet.Client/pull/2884/files#diff-1e206d15ba9babb40ef77ee2b4e4b544R48).  The second one was signed.

The fix was to [copy the first ZIP](https://github.com/NuGet/NuGet.Client/pull/2884/files#diff-1e206d15ba9babb40ef77ee2b4e4b544R87) and then sign the copy.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**